bigquery/google/cloud/bigquery/job.py

```diff
@@ -92,6 +93,14 @@ def _error_result_to_exception(error_result):
     )
 
 
+def _contains_order_by(query):
+    """Do we need to preserve the order of the query results?"""
+    if query is None:
+        return False
+
+    return re.search(r"ORDER\s+BY", query, re.IGNORECASE) is not None
```

Comment:
Should we have a comment indicating we know there's a class of false positives (ordered window functions)?